src/rpc/mining.cpp

```diff
@@ -47,7 +47,7 @@
  * or from the last difficulty change if 'lookup' is nonpositive.
  * If 'height' is nonnegative, compute the estimate at the time when a given block was found.
  */
-static UniValue GetNetworkHashPS(int lookup, int height, const CChain& active_chain) {
+static UniValue GetNetworkHashPS(int lookup, int height, const CChain& active_chain, int algo) {
```

Comment:
@cdonnachie ,

How does this impact RPC callers who do not pass this new variable?  Will this cause any backward compatibility issues?  Should we consider adding an overloaded version of this method so that it remains backward compatible?

Reply:
There is no difference to the way this works.  The algo is optional an will default to what algo the wallet is running under.  There was an issue with the index number it was checking which I have corrected.  
If run with no params is will run with the same defaults 120 -1 current_algo

Reply:
@gto90 this not a new variable, this RPC call already working at 7.17.2 and someone missing to add it to 8.22.0.  This is definitely should be merged.  When some pool has multiple mining algos support, it should not require 5(x2) daemons for sure.

Reply:
Thank you @dgb256 !